dask/dataframe/core.py

```diff
@@ -4555,14 +4556,80 @@ def sort_values(
             **kwargs,
         )
 
+    def _check_index(self, other: Any) -> Optional[Any]:
```

Comment:
Do the type annotations help here? I think there's an existing effort to annotate the dask/dask codebase, see #8854

Reply:
As a rule of thumb, I always add annotations in the code I write, so that it's not another contributor that has to figure out the annotation types later. In this particular case maybe it's not worth it, as `other` can be of several types, but in `set_index` I think it's definitely worth it.

Reply:
I've added more type annotations to several methods in `shuffle.py` as a bonus.